csharp/src/Ice/SocketStream.cs

```diff
@@ -383,6 +401,20 @@ internal virtual async ValueTask SendInitializeFrameAsync(CancellationToken canc
             // The default implementation only supports the Ice2 protocol
             Debug.Assert(_socket.Endpoint.Protocol == Protocol.Ice2);
 
+            if (frame.Size > _socket.OutgoingFrameMaxSize)
+            {
+                if (frame is OutgoingRequestFrame)
+                {
+                    throw new MarshalException($"the request size is larger than {_socket.OutgoingFrameMaxSize}");
+                }
+                else
+                {
+                    // Throw a remote exception instead for response, the Ice connection will catch it and send it
+                    // as the response instead of sending this response which is too large.
+                    throw new MarshalException($"the response size is larger than {_socket.OutgoingFrameMaxSize}");
```

Comment:
For the error message will be something like `the response size is larger than 1048576`?

I am concerned it won't be clear to the reader where this number of coming from. `the response size is larger than the caller's IncomingFrameMaxSize (1048576 bytes)`?

Reply:
Fixed to `$@"the request size is larger than the peer's IncomingFrameSizeMax ({_socket.PeerIncomingFrameMaxSize} bytes)'`